controllers/common.go

```diff
@@ -92,3 +102,35 @@ func reconcileDesiredObjects(ctx context.Context, kubeClient client.Client, logg
 	}
 	return nil
 }
+
+func enabledAcceleratedComputeByAgentConfig(ctx context.Context, c client.Client, log logr.Logger) bool {
```

Comment:
nit: rename to `acceleratedComputeEnabled(config *CwaConfig, log logr.Logger)` and have the actual config struct passed in so that it's easier to test and have the caller get the ACWA resource.